Sources/IBLinterKit/Rules/UseBaseClassRule.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  UseBaseClassRule.swift
+//  IBLinterKit
+//
+//  Created by masamichi on 2019/03/07.
+//
+
+import Foundation
+import IBDecodable
+
+private extension XibFile {
+    var fileExtension: String {
+        return URL.init(fileURLWithPath: pathString).pathExtension
+    }
+    var fileNameWithoutExtension: String {
+        return fileName.replacingOccurrences(of: ".\(fileExtension)", with: "")
+    }
+}
+
+private extension StoryboardFile {
+    var fileExtension: String {
+        return URL.init(fileURLWithPath: pathString).pathExtension
+    }
+    var fileNameWithoutExtension: String {
+        return fileName.replacingOccurrences(of: ".\(fileExtension)", with: "")
+    }
+}
+
+extension Rules {
+    public struct UseBaseClassRule: Rule {
+
+        public static var identifier: String = "use_base_class"
+
+        private var baseClasses: [String: [String]] = [:]
+
+        public init(context: Context) {
+            for baseClassConfig in context.config.useBaseClassRule {
+                self.baseClasses[baseClassConfig.elementClass] = baseClassConfig.baseClasses
+            }
+        }
+
+        public func validate(storyboard: StoryboardFile) -> [Violation] {
+            guard let scenes = storyboard.document.scenes else { return [] }
+            let views = scenes.compactMap { $0.viewController?.viewController.rootView }
+            return views.flatMap { validate(for: $0, file: storyboard, fileNameWithoutExtension: storyboard.fileNameWithoutExtension) }
+        }
+
+        public func validate(xib: XibFile) -> [Violation] {
+            guard let views = xib.document.views else { return [] }
+            return views.flatMap { validate(for: $0.view, file: xib, fileNameWithoutExtension: xib.fileNameWithoutExtension) }
+        }
+
+        private func validate<T: InterfaceBuilderFile>(for view: ViewProtocol, file: T, fileNameWithoutExtension: String) -> [Violation] {
```

Comment:
Is `fileNameWithoutExtension` needed?

Reply:
Oh, it's not used anywhere😅 I can remove it!